core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -30,3 +30,12 @@ class SparkException(message: String, cause: Throwable)
  */
 private[spark] class SparkDriverExecutionException(cause: Throwable)
   extends SparkException("Execution error", cause)
+
+/**
+ * Exception indicating an error internal to Spark, not caused by user error
+ */
+class SparkInternalStateException(message: String, cause: Throwable)
```

Comment:
yeah, I wasn't sure what to name it -- I wanted the name to clearly indicate to the user that the error wasn't their fault, and there isn't anything they can do about it.  I guess your version is better.